WalletWasabi.Tests/UnitTests/JsonConverters/MoneySatoshiJsonConverterTests.cs

```diff
@@ -0,0 +1,189 @@
+using NBitcoin;
+using WalletWasabi.Helpers;
+using WalletWasabi.JsonConverters.Bitcoin;
+using Xunit;
+using JsonConvertNew = System.Text.Json.JsonSerializer;
+using JsonConvertOld = Newtonsoft.Json.JsonConvert;
+
+namespace WalletWasabi.Tests.UnitTests.JsonConverters;
+
+public class MoneySatoshiJsonConverterTests
+{
+	/// <summary>
+	/// Tests that JSON converter based on <c>System.Text.Json</c> and the one based on <c>NewtonSoft.Json</c> *serialize* objects equally.
+	/// </summary>
+	[Fact]
+	public void SerializationParity()
+	{
+		TestData testObject = new();
+
+		string json = AssertSerializedEqually(testObject);
+		Assert.Equal("""{"OneCoin":100000000,"OneSatoshi":1,"Zeros":100,"Max":2099999997690000,"None":null,"NotAnnotated":null}""", json);
+	}
+
+	/// <summary>
+	/// Tests that JSON converter based on <c>System.Text.Json</c> and the one based on <c>NewtonSoft.Json</c> *deserialize* objects equally.
+	/// </summary>
+	[Fact]
+	public void DeserializationParity()
+	{
+		// Success cases.
+		{
+			string token = "2100000000000000"; // 21 million.
+			AssertBothDeserialize(token);
+		}
+
+		// Json exceptions.
+		// Newtonsoft gives JsonReaderException, Microsoft gives JsonException. Similar but not the same.
+		{
+			string token = "1,000.00"; // Thousand separator.
+			AssertDeserializeJsonException(token);
+
+			token = "1,0"; // Decimal comma.
+			AssertDeserializeJsonException(token);
+		}
+
+		// Casting errors Vs Format Exception.
+		{
+			// Newtonsoft gives back InvalidCastException, Microsoft gives back FormatException.
+			string token = "0.00000000000000000000000000000000000000000000001";
+			AssertDeserializeFailure<InvalidCastException>(token);
+
+			// Newtonsoft gives back InvalidCastException, Microsoft gives back FormatException.
+			token = "209999999.97690001"; // Maximum number of bitcoins ever to exist + 1 satoshi.
+			AssertDeserializeFailure<InvalidCastException>(token);
+
+			// Newtonsoft gives back InvalidCastException, Microsoft gives back FormatException.
+			token = "1e6"; // Exponential notation.
+			AssertDeserializeFailure<InvalidCastException>(token);
```

Comment:
`FormatException` is thrown [here](https://github.com/zkSNACKs/WalletWasabi/pull/11840/files#diff-3de5a36861e4beb186c7b6fd21816b0d57f4207faf44263eb1f867da6393027aR17).